### ISSUE TITLE
Implementation of SpatialDropout seems incorrect

### ISSUE THREAD

**Issue by JoshVarty** [JoshVarty/GoogleQALabeling#6]
Implementation:

```
class SpatialDropout(nn.Dropout2d):
    def forward(self, x):
        x = x.unsqueeze(2)  # (N, T, 1, K)
        x = x.permute(0, 3, 2, 1)  # (N, K, 1, T)
        x = super(SpatialDropout, self).forward(x)  # (N, K, 1, T), some features are masked
        x = x.permute(0, 3, 2, 1)  # (N, T, 1, K)
        x = x.squeeze(2)  # (N, T, K
```

I think we're supposed to randomly dropout entire channels NOT entire words. I think this implementation ends up dropping entire words.

We should check both of these statements ^

**Comment by JoshVarty** [comment]
Nope it's correct. When `T=500` (we have 500 words) we knock out 500 elements, one feature for each word. To double check:


```Python
import torch
import torch.nn as nn

class SpatialDropout(nn.Dropout2d):
    def forward(self, x):
        x = x.unsqueeze(2)    # (N, T, 1, K)
        x = x.permute(0, 3, 2, 1)  # (N, K, 1, T)
        x = super(SpatialDropout, self).forward(x)  # (N, K, 1, T), some features are masked
        x = x.permute(0, 3, 2, 1)  # (N, T, 1, K)
        x = x.squeeze(2)  # (N, T, K)
        return x

dropout = SpatialDropout()
embedding = torch.randn(1, 5, 3)	# 1 batch, 5 words, feature length of 3
print(embedding)
print(dropout(embedding))
```
